components/jira/jira.app.mjs

```diff
@@ -1,11 +1,257 @@
+import { axios } from "@pipedream/platform";
+
 export default {
   type: "app",
   app: "jira",
-  propDefinitions: {},
+  propDefinitions: {
+    projectID: {
+      type: "string",
+      label: "Project ID",
+      description: "The project ID.",
+      async options() {
+        const resp = await this._makeRequest({
+          method: "GET",
+          path: "/project",
+        });
+        return resp.map((e) => ({
+          label: e.name,
+          value: e.id,
+        }));
+      },
+    },
+    issueType: {
+      type: "string",
+      label: "Issue Type",
+      description: "An ID identifying the type of issue, [Check the API docs](https://developer.atlassian.com/cloud/jira/platform/rest/v3/#api-rest-api-3-issue-post) to see available options",
+      async options({ projectID }) {
+        const resp = await this._makeRequest({
+          method: "GET",
+          path: "/issue/createmeta",
+          params: {
+            projectIds: projectID,
+          },
+        });
+        // Because we select a project, we get issue types for only that and only one project
+        return resp.projects[0].issuetypes.map((e) => ({
+          label: e.name,
+          value: e.id,
+        }));
+      },
+    },
+    issueIdOrKey: {
+      type: "string",
+      label: "Issue id or key",
+      description: "The ID or key of the issue where the attachment will be added to.",
+    },
+    accountId: {
+      type: "string",
+      label: "Account Id",
+      description: "The account ID of the user, which uniquely identifies the user across all Atlassian products, For example, `5b10ac8d82e05b22cc7d4ef5`, ",
+    },
```

Comment:
I don't think having a prop for `accountId` is necessary. We can retrieve the authenticated user's accountId from the API:
https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-myself/#api-rest-api-3-myself-get

Reply:
This one is actually used for not users themselves but other users who can be assigned to an issue. For example, user A assigns an issue to user B, and this accountId is used for user B in this case. How about a paginated option prop for users here?